croutonjs/src/js/crouton-core.js

```diff
@@ -237,9 +237,9 @@ function Crouton(config) {
 		var uniqueIds = arrayUnique(idsArray);
 		jQuery.getJSON(self.config['root_server'] + '/client_interface/jsonp/?switcher=GetFormats&callback=?', function (data) {
 			var formats = [];
-			for (var format of data) {
-				if (inArray(format['id'], uniqueIds)) {
-					formats.push(format);
+			for (var i = 0; i < data.length; i++) {
```

Comment:
For readability in these types of `for` loops, I like to pull the value out of the array and store it in a variable the indicates its name.


```
for (var i = 0; i < data.length; i++) {
    var format = data[i];
    formats.push(format);
}
```